add function to include data attributes

Roilan opened the pull request:
React allows custom `data` attributes. I added a simple function to spread an object that contains a `data-` prop. If you have a better idea or way to implementing this, feel free to share.

chromakode commented:
Code looks good to me, thanks @Roilan! I'm curious, what use case do you have for data props in emails?

Roilan commented:
@chromakode We're actually using the compiled templates and tossing them into Mailchimp. Mailchimp has these custom `mc:name` attributes that you specify and allows you to make changes via Mailchimp's design view (drag/drop, edit text, etc.) React won't accept `mc:` attributes, so I will be adding `data-mc-name` and doing `.replace` to insert `mc:name`.

Roilan commented:
@chromakode I just added a bit more to this PR. It adds an option to `renderEmail` for replacing `data-mc-` with Mailchimp's custom `mc:` attribute that I mentioned earlier. 

I also added an additional test and made sure the existing test pass without any issues. Take a look over and see if you spot anything or have suggestions for improvements.

chromakode commented on src/components/Box.js:
```diff
@@ -1,9 +1,10 @@
 export default function Box({ children, ...props }) {
   return (
-    <table {...props}>
+    <table {...includeDataProps(props)} {...props}>
```
Is `includeDataProps` redundant here, since we're already passing through all `props`?

Roilan replied:
Ah, yes. Good catch.

chromakode commented on src/renderEmail.js:
```diff
@@ -1,6 +1,12 @@
-  return doctype + ReactDOMServer.renderToStaticMarkup(emailComponent)
+  let markup = doctype + ReactDOMServer.renderToStaticMarkup(emailComponent)
+
+  if (renderMailchimpAttr) {
```
I would prefer for `react-html-email` to not include vendor-specific functionality. This would also be useful to users outside of React. What do you think of breaking out the data attribute -> `mc:` transform into a separate standalone utility?

Roilan replied:
Understood and that makes sense. How do you feel about utils we can wrap around our markup string?

ie: renderMailChimpAttr( renderEmail(<Component />) )

chromakode replied:
I think the MailChimp data attribute transform is really useful, but it's a separate concern from HTML email generation. I want to keep the API surface area for this library as small and generic as possible. Two ideas:

* I wonder if there's a way to get React to pass these attributes through, [`injectReactEmailAttributes`](https://github.com/chromakode/react-html-email/blob/master/src/injectReactEmailAttributes.js) style?

Alternatively:

* How about you create your own "mailchimpify" module which contains this transform, which we link to in the documentation?

Roilan replied:
Great idea. I think your suggestion to have it as it's own module would be ideal. It's such a small change that anyone else would be able to use outside of react-html-email and might find it useful.

chromakode commented on src/components/A.js:
```diff
@@ -1,9 +1,10 @@
 import React, { PropTypes } from 'react'
 import EmailPropTypes from '../PropTypes'
+import includeDataProps from '../includeDataProp'
```
Small nit here: could you please make the filename of the module match the import name, `includeDataProps`?

Roilan replied:
Another good catch, I'll make the file name adjustment.

chromakode commented:
This looks great @Roilan, thanks! Just a few notes/nits, then let's get this merged and roll a new release :)

Roilan commented:
@chromakode Made changes based on your feedback. Thanks for the review 👍

chromakode commented on src/includeDataProps.js:
```diff
@@ -0,0 +1,11 @@
+  const dataProps = {}
+
+  Object.keys(props).forEach(key => {
+    if (key.indexOf('data-') > -1) {
```
Just noticed this will match strings like `foo-data-bar`, which aren't valid data attributes. What do you think of changing the predicate to `/^data-/.test(key)`?

Roilan replied:
Ah, yes. You're right. It should only match strings that start with `data-`. I'll make this change.

chromakode replied:
Thanks!

chromakode commented on src/renderEmail.js:
```diff
@@ -2,5 +2,6 @@ import ReactDOMServer from 'react-dom/server'
 
 export default function renderEmail(emailComponent) {
   const doctype = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">'
+
```
I think this blank line snuck in after your last commit :scissors:

Roilan commented:
@chromakode Thanks for the additional feedback. Updated. I can throw in another test for the data props if you'd like.

chromakode commented:
Yay! Thanks for the fast fixes. I'm going to add a quick fix to #5 and call this 1.0

Would you like to join as a maintainer of this project?

chromakode commented:
Tethering seems to be rejecting my `npm` PUT requests and my VPN is currently unhappy -- will have to wait a few hours to roll this out.

Roilan commented:
Ah, no worries. No rush on my end. That would be awesome. Found this module super helpful and I can only see myself adding more to it as I need it and others report issues.

chromakode commented:
Yeah especially if you're going to be using it in production, you're a primary stakeholder!

Roilan commented:
Yeah, probably more ideas will come up as I keep building this template / component system for emails with react-html-email.

chromakode commented:
Done! :raised_hands:
